contracts/cw-core/src/contract.rs

```diff
@@ -312,15 +317,38 @@ pub fn execute_update_proposal_modules(
     env: Env,
     sender: Addr,
     to_add: Vec<ModuleInstantiateInfo>,
-    to_remove: Vec<String>,
+    to_disable: Vec<String>,
 ) -> Result<Response, ContractError> {
     if env.contract.address != sender {
         return Err(ContractError::Unauthorized {});
     }
 
-    for addr in to_remove {
+    for addr in to_disable {
         let addr = deps.api.addr_validate(&addr)?;
-        PROPOSAL_MODULES.remove(deps.storage, addr);
+        let mut module = PROPOSAL_MODULES
+            .load(deps.storage, addr.clone())
+            .map_err(|_| ContractError::ProposalModuleDoesNotExist {
+                address: addr.clone(),
+            })?;
+
+        if module.status == ProposalModuleStatus::Disabled {
+            return Err(ContractError::ModuleAlreadyDisabled {
+                address: module.address,
+            });
+        }
+
+        // If disabling this module will cause there to be no active modules, return error.
+        // We don't check the active count before disabling because there may erroneously be
+        // modules in to_disable which are already disabled.
+        ACTIVE_PROPOSAL_MODULE_COUNT.update(deps.storage, |count| {
+            if count <= 1 && to_add.is_empty() {
+                return Err(ContractError::NoActiveProposalModules {});
+            }
+            Ok(count - 1)
+        })?;
```

Comment:
How would you feel about doing this at the end of this for loop based on the length of `to_disable`? Stops us from doing a storage operation every time.

Reply:
much better idea haha